### PR TITLE
Update packaging script to build reference assembly targeting `netcoreapp2.1` and use actual `.csproj` files

### LINKED ISSUE #8721: Check in the .csproj files that are used to build reference assemblies for our SDK nuget packages
# Summary of the new feature/enhancement

Currently the `.csproj` files for build reference assemblies are generated on-the-fly (the content is kept in a resource.psd1 file as templates).
We need a way to update the version of nuget packages referenced by those `.csproj` files, and the proposal is to have the .csproj files checked in, and then we can use the DependBot to manage the version of nuget packages referenced by those .csproj files.

Also, the generated SDK nuget packages are currently declared as `netstandard2.0`. We should change it to `netcoreapp2.1` as the runtime libraries are `netcoreapp2.1` specific.

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Update packaging script to build reference assembly targeting `netcoreapp2.1` and use actual `.csproj` files.
Fix #8721

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/packaging/projects/reference/System.Management.Automation.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <Version>$(RefAsmVersion)</Version>
+    <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>$(SnkFile)</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <LangVersion>Latest</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Management.Infrastructure" Version="1.0.0" />
+    <PackageReference Include="System.Security.AccessControl" Version="4.4.1" />
+    <PackageReference Include="System.Security.Principal.Windows" Version="4.4.1" />
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply tools/packaging/projects/reference/System.Management.Automation.csproj]: Fixed.

File: tools/packaging/projects/reference/System.Management.Automation.csproj [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <Version>$(RefAsmVersion)</Version>
+    <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>$(SnkFile)</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <LangVersion>Latest</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Management.Infrastructure" Version="1.0.0" />
+    <PackageReference Include="System.Security.AccessControl" Version="4.4.1" />
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply tools/packaging/projects/reference/System.Management.Automation.csproj]: Fixed.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by daxian-dbw** [comment]
Verified by build https://mscodehub.visualstudio.com/PowerShellCore/_build%2Fresults?buildId=72005&_a=summary
Build passed and the generated nuget packages are also manually verified.

**Comment by TravisEz13** [comment]
We may need to tell dependabot to rescan the repo after this is merged.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**